libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/Span.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing;
+
+/**
+ * An interface that represents a tracing span.
+ * Spans are created by the Tracer.startSpan method.
+ * Span must be ended by calling Tracer.endSpan which internally calls Span's endSpan.
+ */
+public interface Span {
```

Comment:
Should Span implement `Closeable` as well?

Reply:
It could and the `close` would call `endSpan` method but we dont want to expose `Span` to users to ensure they dont operate on `Span` directly. Instead we want all span operations proxied through Tracer. This was needed because we maintain state (of current active span) in Tracer. If the users end the span directly. our framework will not be aware of the span end and hence will result in issues.

Reply:
Fair, let's do not expose `Span` but lighter abstraction like `Scope` for example: it will implement `Closeable ` / `AutoCloseable` and provide more easier APIs over attributes / baggage (in future) management.

Reply:
Makes sense, let me try

Reply:
Done in latest commit